### ISSUE TITLE
Fast create & delete of vdisk via api causes leftover data

### ISSUE THREAD

**Issue by kinvaris** [openvstorage/framework#1114]
When we looked at the CI tests this morning we saw that although `VDiskController.delete` had not given an error, there was still leftover data in the model of a vdisk.

You can easily reproduce it via this piece of code (on a environment where CI is installed):
```
from ci.remove.vdisk import VDiskRemover
from ci.setup.vdisk import VDiskSetup
from ci.helpers.api import OVSClient
import json

VDISK_SIZE = 1073741824
CONFIG_LOC = "/opt/OpenvStorage/ci/config/setup.json"

with open(CONFIG_LOC, "r") as JSON_CONFIG:
    config = json.load(JSON_CONFIG)
api = OVSClient(
    config['ci']['grid_ip'],
    config['ci']['user']['api']['username'],
    config['ci']['user']['api']['password']
)

for nr in xrange(1000):
    print VDiskSetup.create_vdisk(vdisk_name="jonas-testc{0}.raw".format(nr), vpool_name="myvpool01", size=VDISK_SIZE, storagerouter_ip="10.100.199.191", api=api)
    print VDiskRemover.remove_vdisk_by_name("jonas-testc{0}.raw".format(nr), "myvpool01")
```

As seen here: 
![](https://i.imagebanana.com/img/4sqi66aa/Selection_008.png)

**Comment by kinvaris** [comment]
After a second look I saw this:

Deletion occured, disk is not found anymore as seen here:

```
jonas-testd9.raw <-- creation
/usr/local/lib/python2.7/dist-packages/requests/packages/urllib3/connectionpool.py:791: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html
  InsecureRequestWarning)
/usr/local/lib/python2.7/dist-packages/requests/packages/urllib3/connectionpool.py:791: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html
  InsecureRequestWarning)
/usr/local/lib/python2.7/dist-packages/requests/packages/urllib3/connectionpool.py:791: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html
  InsecureRequestWarning)
5c6afef6-a3c2-4d59-ad22-c50e58a219f0 <-- deletion
exception! VDisk with guid '5c6afef6-a3c2-4d59-ad22-c50e58a219f0' could not be found <-- deleted?
True <-- result of deletion!
```

But when looking at the vdisk guid, we see it is now different than previously seen in the model! It should be: `5c6afef6-a3c2-4d59-ad22-c50e58a219f0` but now its:

```
In [12]: v.name
Out[12]: u'jonas-testd9'

In [13]: v.guid
Out[13]: '3c70a545-646a-4a98-af86-ab3d5c50110f'

```

**Comment by khenderick** [comment]
Fixed by #1117, packaged in openvstorage-2.7.4-rev.4258.c9df8b4

**Comment by kinvaris** [comment]
PASSED: the healthcheck and integration tests are not producing any empty disks anymore on our CI environments.
